purchases/src/androidTestDefaults/kotlin/com/revenuecat/purchases/PurchasesIntegrationTest.kt

```diff
@@ -205,6 +208,170 @@ class PurchasesIntegrationTest : BasePurchasesIntegrationTest() {
+                    },
+                )
+            },
+        )
+
+        lock.await(testTimeout.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    @Test
+    fun testGettingVirtualCurrenciesForNewUserReturnsVCsWith0Balance() {
+        val newAppUserID = "integrationTestUser_${UUID.randomUUID()}"
+        val lock = CountDownLatch(1)
+
+        Purchases.sharedInstance.logInWith(
+            appUserID = newAppUserID,
+            onError = { error -> fail("should have been able to login. Error: $error") },
+            onSuccess = { _, created ->
+                assertThat(created).isTrue() // This user should be new
+
+                Purchases.sharedInstance.invalidateVirtualCurrenciesCache()
+
+                Purchases.sharedInstance.getVirtualCurrenciesWith(
+                    onError = { error -> fail("should be success. Error: $error") },
+                    onSuccess = { virtualCurrencies ->
+                        validateAllZeroBalanceVirtualCurrenciesObject(virtualCurrencies = virtualCurrencies)
+                        lock.countDown()
+                    },
+                )
+            },
+        )
+
+        lock.await(testTimeout.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    @Test
+    fun testCachedVirtualCurrencies() {
+        val appUserID = "integrationTestUserWithAllBalancesNonZero"
+        val lock = CountDownLatch(1)
+
+        Purchases.sharedInstance.logInWith(
+            appUserID = appUserID,
+            onError = { error -> fail("should have been able to login. Error: $error") },
+            onSuccess = { _, created ->
+                assertThat(created).isFalse() // This user should be already exist
+
+                Purchases.sharedInstance.invalidateVirtualCurrenciesCache()
+
+                Purchases.sharedInstance.getVirtualCurrenciesWith(
+                    onError = { error -> fail("should be success. Error: $error") },
+                    onSuccess = { virtualCurrencies ->
+                        validateAllNonZeroBalanceVirtualCurrenciesObject(virtualCurrencies = virtualCurrencies)
+
+                        var cachedVirtualCurrencies = Purchases.sharedInstance.cachedVirtualCurrencies
+                        validateAllNonZeroBalanceVirtualCurrenciesObject(virtualCurrencies = cachedVirtualCurrencies)
+
+                        Purchases.sharedInstance.invalidateVirtualCurrenciesCache()
+                        cachedVirtualCurrencies = Purchases.sharedInstance.cachedVirtualCurrencies
+                        assertThat(cachedVirtualCurrencies).isNull()
+
+                        lock.countDown()
+                    },
+                )
+            },
+        )
+
+        lock.await(testTimeout.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    private fun validateAllZeroBalanceVirtualCurrenciesObject(virtualCurrencies: VirtualCurrencies?) {
```

Comment:
Nitpick but I think naming can be simplified like this considering the context? 
```suggestion
    private fun validateAllZeroBalances(virtualCurrencies: VirtualCurrencies?) {
```
Same for the method above

Reply:
Great idea! Modified the names here 3ec729ca08d11052896c49f2665cd41cc76cdc9f 👍